robot_sf/render/sim_view.py

```diff
@@ -427,25 +427,23 @@ def _draw_coordinates(self, x, y):
         text = self.font.render(f'({x}, {y})', False, TEXT_COLOR)
         self.screen.blit(text, (x, y))
 
-    def _augment_timestep(self, timestep: int):
-        # TODO: show map name as well
-        text = f'step: {timestep}'
-        text_surface = self.font.render(text, False, TEXT_COLOR)
-        self.screen.blit(text_surface, self._timestep_text_pos)
-
     def _add_text(self, timestep: int, state: VisualizableSimState):
-        if self.display_robot_info:
-            lines = [
+        lines = []
+        if self.display_robot_info == 1 and state.robot_action:
+            lines += [
                 f'RobotPose: {state.robot_pose}',
-                f'RobotAction: {state.robot_action.action}',
-                f'RobotGoal: {state.robot_action.goal}']
-        else:
-            distance_to_robot = euclid_dist(state.ego_ped_pose[0], state.robot_pose[0])
-            lines = [
-                f'PedestrianPose: {state.ego_ped_pose}',
-                f'PedestrianAction: {state.ego_ped_actions.action}',
-                f'PedestrianGoal: {state.ego_ped_actions.goal}',
-                f'DistanceRobot: {distance_to_robot:.2f}',]
+                f'RobotAction: {state.robot_action.action if state.robot_action else None}',
+                f'RobotGoal: {state.robot_action.goal if state.robot_action else None}',]
+        elif self.display_robot_info == 2:
+            if state.ego_ped_pose and state.ego_ped_action:
+                distance_to_robot = euclid_dist(state.ego_ped_pose[0], state.robot_pose[0])
+                lines += [
+                    f'PedestrianPose: {state.ego_ped_pose}',
+                    f'PedestrianAction: {state.ego_ped_action.action}',
+                    f'PedestrianGoal: {state.ego_ped_action.goal}',
+                    f'DistanceRobot: {distance_to_robot:.2f}',]
+            else:
+                self.display_robot_info = 0
```

Comment:
With your experience @JuliusMiller, why did you not use this suggestion?

Reply:
In my opinion it is a better user experience to switch between no_info and robot_info, if the pedestrian is not present.
 
Otherwise an extra click would be required, which can be frustrating for people who are only interested in the robot.